storage/src/nodestore/hash.rs

```diff
@@ -33,99 +79,95 @@ where
         let mut nodes_processed = 1usize; // Count this node
         if let Node::Branch(ref mut b) = node {
-            // special case code for ethereum hashes at the account level
             #[cfg(feature = "ethhash")]
-            let make_fake_root = if path_prefix.0.len().saturating_add(b.partial_path.0.len()) == 64
-            {
-                // looks like we're at an account branch
-                // tally up how many hashes we need to deal with
+            // special case for ethhash at the account level
+            let num_children = if path_prefix.0.len().saturating_add(b.partial_path.0.len()) == 64 {
                 let ClassifiedChildren {
-                    unhashed,
+                    num_unhashed,
                     mut hashed,
-                } = self.ethhash_classify_children(&mut b.children);
-                trace!("hashed {hashed:?} unhashed {unhashed:?}");
-                if hashed.len() == 1 {
-                    // we were left with one hashed node that must be rehashed
-                    let (invalidated_node_idx, (invalidated_node, invalidated_hash)) =
-                        hashed.first_mut().expect("hashed is not empty");
-                    // Extract the address from the MaybePersistedNode
-                    let addr: crate::LinearAddress = invalidated_node
-                        .as_linear_address()
-                        .expect("hashed node should be persisted");
-                    let mut hashable_node = self.read_node(addr)?.deref().clone();
-                    let original_length = path_prefix.len();
-                    path_prefix.0.extend(b.partial_path.0.iter().copied());
-                    if unhashed.is_empty() {
-                        hashable_node.update_partial_path(Path::from_nibbles_iterator(
-                            std::iter::once(*invalidated_node_idx as u8)
-                                .chain(hashable_node.partial_path().0.iter().copied()),
-                        ));
-                    } else {
-                        path_prefix.0.push(*invalidated_node_idx as u8);
-                    }
-                    let hash = hash_node(&hashable_node, path_prefix);
-                    path_prefix.0.truncate(original_length);
-                    **invalidated_hash = hash;
-                }
-                // handle the single-child case for an account special below
-                if hashed.is_empty() && unhashed.len() == 1 {
-                    Some(unhashed.last().expect("only one").0 as u8)
-                } else {
-                    None
+                } = self.ethhash_classify_children(&mut b.children)?;
+                trace!("hashed {hashed:?} unhashed {num_unhashed:?}");
+                #[expect(
+                    clippy::arithmetic_side_effects,
+                    reason = "hashed and unhashed can have at most 16 elements"
+                )]
+                let num_children = hashed.len() + num_unhashed;
```

Comment:
nit: just use wrapping_add, which is free. The comment can stay as to why if you want.
```suggestion
                // this sum can never exceed 16
                let num_children = hashed.len().wrapping_add(num_unhashed);
```